service/controller/app/v1/resource/chart/current.go

```diff
@@ -19,6 +20,20 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 		return nil, microerror.Mask(err)
 	}
 
+	ns, err := r.k8sClient.CoreV1().Namespaces().Get(cr.Namespace, metav1.GetOptions{})
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	if ns.GetDeletionTimestamp() != nil {
```

Comment:
Brainwave, what we can do is to have a resource that gathers information in the controller context which other resources then make use of. So we could have a resource to fetch the namespace and put it into the controller context or directly have it put the boolean. So this here would then only be something like this. 

```
if cc.Status.TenantCluster.Namespace.IsDeleting {
```

We make use of this concept in a lot of places already. See for instance the following. 

* https://github.com/giantswarm/aws-operator/tree/master/service/controller/clusterapi/v29/resource/accountid
* https://github.com/giantswarm/aws-operator/tree/master/service/controller/clusterapi/v29/resource/awsclient
* https://github.com/giantswarm/aws-operator/tree/master/service/controller/clusterapi/v29/resource/region

Reply:
Thanks for the reference! This is great, I think new resource should be located in the first place in the resource_set, right?